Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -87,13 +95,47 @@ public sealed class Server : ServerBase, IDisposable {
         #region Client message handling
 
         public override async Task<InitializeResult> Initialize(InitializeParams @params) {
-            _testEnvironment = @params.initializationOptions.interpreter.properties.ContainsKey("TestEnvironment");
-            if (_testEnvironment) {
-                // Test environment needs predictable initialization.
-                // Tests can only proceed when analysis is fully done.
-                _analyzer = CreateAnalyzer(@params.initializationOptions.interpreter).Result;
+            _testEnvironment = @params.initializationOptions.testEnvironment;
+            // Test environment needs predictable initialization.
+            if (@params.initializationOptions.asyncStartup && !_testEnvironment) {
+                CreateAnalyzer(@params.initializationOptions.interpreter).ContinueWith(t => {
+                    if (t.IsFaulted) {
+                        _analyzerCreationTcs.TrySetException(t.Exception);
+                    } else {
+                        try {
+                            _analyzer = t.Result;
+                            OnAnalyzerCreated(@params);
+                            _analyzerCreationTcs.TrySetResult(true);
+                        } catch (Exception ex) {
+                            _analyzerCreationTcs.TrySetException(ex);
+                        }
+                    }
+                }).DoNotWait();
             } else {
                 _analyzer = await CreateAnalyzer(@params.initializationOptions.interpreter);
+                OnAnalyzerCreated(@params);
+                _analyzerCreationTcs.TrySetResult(true);
```

Comment:
Behavior here is different from when `@params.initializationOptions.asyncStartup && !_testEnvironment` is true. If `CreateAnalyzer` raises an exception, `_analyzerCreationTcs` will not complete. Is it by design?

Reply:
Let me try-catch that. Although if it throws, nothing is going to work, but at least exception may give client an idea that server may need to be restarted.